includes/gateways/cod/class-wc-gateway-cod.php

```diff
@@ -63,42 +65,46 @@ protected function setup_properties() {
 	 */
 	public function init_form_fields() {
 
-		$options    = array();
-		$data_store = WC_Data_Store::load( 'shipping-zone' );
-		$raw_zones  = $data_store->get_zones();
+		$options = array();
 
-		foreach ( $raw_zones as $raw_zone ) {
-			$zones[] = new WC_Shipping_Zone( $raw_zone );
-		}
+		// We only want to load the options if they'll be shown to the user.
+		if ( $this->is_accessing_settings() ) {
```

Comment:
Since this method sets the `$form_fields` property (which is public), I wanted to fix the bug in a way that causes minimal breakage. With this we only remove the `options` content from a single field.

Reply:
I've gone ahead and extracted the functionality into its own method. This should be more readable now!